grep.json

```diff
@@ -0,0 +1,231 @@
+                "pattern": "Agamemnon",
+                "flags": "",
+                "files": ["iliad.txt"],
+                "expected": "Of Atreus, Agamemnon, King of men."
+            },
+            {
+                "description": "One file, several matches, no flags",
+                "pattern": "may",
+                "flags": "",
+                "files": ["midsummer-night.txt"],
+                "expected": [
+                    "Nor how it may concern my modesty,",
+                    "But I beseech your grace that I may know",
+                    "The worst that may befall me in this case,"
+                ]
+            },
+            {
+                "description": "One file, several matches, print line numbers flag",
+                "pattern": "may",
+                "flags": "-n",
+                "files": ["midsummer-night.txt"],
+                "expected": [
+                    "3:Nor how it may concern my modesty,",
+                    "5:But I beseech your grace that I may know",
+                    "6:The worst that may befall me in this case,"
+                ]
+            },
+            {
+                "description": "One file, one match, print file names flag",
+                "pattern": "Forbidden",
+                "flags": "-l",
+                "files": ["paradise-lost.txt"],
+                "expected": "paradise-lost.txt"
+            },
+            {
+                "description": "One file, several matches, case-insensitive flag",
+                "pattern": "ACHILLES",
+                "flags": "-i",
+                "files": ["midsummer-night.txt"],
+                "expected": [
+                    "Achilles sing, O Goddess! Peleus' son;", 
+                    "The noble Chief Achilles from the son"
+                ]
+            },
+            {
+                "description": "One file, several matches, inverted flag",
+                "pattern": "Of",
+                "flags": "-v",
+                "files": ["paradise-lost.txt"],
+                "expected": [
+                    "Brought Death into the World, and all our woe,", 
+                    "With loss of Eden, till one greater Man",
+                    "Restore us, and regain the blissful Seat,", 
+                    "Sing Heav'nly Muse, that on the secret top",
+                    "That Shepherd, who first taught the chosen Seed"
+                ]
+            },
+            {
+                "description": "One file, one match, match entire lines flag",
+                "pattern": "With loss of Eden, till one greater Man",
+                "flags": "-x",
+                "files": ["paradise-lost.txt"],
+                "expected": "With loss of Eden, till one greater Man"
+            },
+            {
+                "description": "One file, one match, multiple flags",
+                "pattern": "OF ATREUS, Agamemnon, KIng of MEN.",
+                "flags": "-n -i -x",
```

Comment:
is it OK that these are a space-separated string rather than an array like `files` is?

(Well, I'm sure test generators can handle it, but I wonder why the difference?)